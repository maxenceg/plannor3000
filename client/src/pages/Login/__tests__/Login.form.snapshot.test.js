// @flow
import React from 'react';
import { shallow } from 'enzyme';

import InnerLoginForm from '../Login.form';

describe('The form component', () => {
  it('should render correctly with no error', () => {
    const props = { errors: {}, touched: {}, isSubmitting: false };
    const tree = shallow(<InnerLoginForm {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when submitting', () => {
    const props = { errors: {}, touched: {}, isSubmitting: true };
    const tree = shallow(<InnerLoginForm {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with errors', () => {
    const props = { errors: {username: 'Username is required'}, touched: {username: true}, isSubmitting: false };
    const tree = shallow(<InnerLoginForm {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
