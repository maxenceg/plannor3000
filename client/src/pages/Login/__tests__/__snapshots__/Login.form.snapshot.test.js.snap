// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form component should render correctly when submitting 1`] = `
<Form>
  <div>
    <C
      component={[Function]}
      label="Username"
      name="username"
      type="text"
    />
  </div>
  <div>
    <C
      component={[Function]}
      label="Password"
      name="password"
      type="password"
    />
  </div>
  <WithStyles(Button)
    color="primary"
    disabled={true}
    size="medium"
    type="submit"
    variant="outlined"
  >
    Connect
  </WithStyles(Button)>
</Form>
`;

exports[`The form component should render correctly with errors 1`] = `
<Form>
  <div>
    <C
      component={[Function]}
      error="Username is required"
      label="Username"
      name="username"
      type="text"
    />
  </div>
  <div>
    <C
      component={[Function]}
      label="Password"
      name="password"
      type="password"
    />
  </div>
  <WithStyles(Button)
    color="primary"
    disabled={false}
    size="medium"
    type="submit"
    variant="outlined"
  >
    Connect
  </WithStyles(Button)>
</Form>
`;

exports[`The form component should render correctly with no error 1`] = `
<Form>
  <div>
    <C
      component={[Function]}
      label="Username"
      name="username"
      type="text"
    />
  </div>
  <div>
    <C
      component={[Function]}
      label="Password"
      name="password"
      type="password"
    />
  </div>
  <WithStyles(Button)
    color="primary"
    disabled={false}
    size="medium"
    type="submit"
    variant="outlined"
  >
    Connect
  </WithStyles(Button)>
</Form>
`;
